src/Native/Runtime/amd64/InteropThunksHelpers.S

```diff
@@ -4,19 +4,27 @@
 
 .intel_syntax noprefix
 #include <unixasmmacros.inc>
-
-//TODO: Implement these stub method for non-windows
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 8
 
 LEAF_ENTRY RhpCommonStub, _TEXT
-    int 3
+    push_nonvol_reg rdi
```

Comment:
I am only passing one argument, so only preserving rdi. But you are right, even though the C++ function is trivial, it can potentially pollute other argument registers. I will preserve all the argument registers.